From 73d1dbc68fd79e8028931b6b20d9ca9848956145 Mon Sep 17 00:00:00 2001
From: Xinyu Chen <xinyu.chen@freescale.com>
Date: Thu, 27 Dec 2012 11:25:01 +0800
Subject: [PATCH 130/602] ENGR00238355 imx6: refine the auto brightness levels

Refer to Nexus7's auto brightness level configures,
we increase the brightness detection from 8 to 12
levels. Also refine the LCD backlight levels according
to 12 brightness levels.

Signed-off-by: Xinyu Chen <xinyu.chen@freescale.com>
---
 .../frameworks/base/core/res/res/values/config.xml | 64 ++++++++++++++++------
 .../frameworks/base/core/res/res/values/config.xml | 58 ++++++++++++++------
 2 files changed, 90 insertions(+), 32 deletions(-)

diff --git a/sabreauto_6q/overlay/frameworks/base/core/res/res/values/config.xml b/sabreauto_6q/overlay/frameworks/base/core/res/res/values/config.xml
index 21fa3f4..610a3f6 100755
--- a/sabreauto_6q/overlay/frameworks/base/core/res/res/values/config.xml
+++ b/sabreauto_6q/overlay/frameworks/base/core/res/res/values/config.xml
@@ -88,14 +88,18 @@
 
          Must be overridden in platform specific overlays -->
     <integer-array name="config_autoBrightnessLevels">
-        <item>10</item>
-        <item>40</item>
+        <item>5</item>
+        <item>15</item>
+        <item>50</item>
         <item>100</item>
-        <item>225</item>
-        <item>800</item>
-        <item>1600</item>
-        <item>3200</item>
-        <item>16000</item>
+        <item>200</item>
+        <item>400</item>
+        <item>1000</item>
+        <item>2000</item>
+        <item>3000</item>
+        <item>5000</item>
+        <item>10000</item>
+        <item>30000</item>
     </integer-array>
 
     <!-- Array of output values for LCD backlight corresponding to the LUX values
@@ -103,15 +107,19 @@
          than the size of the config_autoBrightnessLevels array.
          This must be overridden in platform specific overlays -->
     <integer-array name="config_autoBrightnessLcdBacklightValues">
-        <item>25</item>
-        <item>80</item>
-        <item>130</item>
-        <item>180</item>
-        <item>230</item>
-        <item>240</item>
-        <item>250</item>
-        <item>255</item>
-        <item>255</item>
+        <item>5</item>    <!-- 0-5 -->
+        <item>20</item>   <!-- 5-15 -->
+        <item>30</item>   <!-- 15-50 -->
+        <item>40</item>   <!-- 50-100 -->
+        <item>50</item>   <!-- 100-200 -->
+        <item>60</item>   <!-- 200-400 -->
+        <item>70</item>   <!-- 400-1000 -->
+        <item>80</item>   <!-- 1000-2000 -->
+        <item>130</item>  <!-- 2000-3000 -->
+        <item>180</item>  <!-- 3000-5000 -->
+        <item>255</item>  <!-- 5000-10000 -->
+        <item>255</item>  <!-- 10000-30000 -->
+        <item>255</item>  <!-- 30000+ -->
     </integer-array>
 
     <!-- Array of output values for button backlight corresponding to the LUX values
@@ -128,6 +136,14 @@
         <item>0</item>
         <item>0</item>
         <item>0</item>
+        <item>0</item>
+        <item>0</item>
+        <item>0</item>
+        <item>0</item>
+        <item>0</item>
+        <item>0</item>
+        <item>0</item>
+        <item>0</item>
     </integer-array>
 
     <!-- Array of output values for keyboard backlight corresponding to the LUX values
@@ -144,8 +160,18 @@
         <item>0</item>
         <item>0</item>
         <item>0</item>
+        <item>0</item>
+        <item>0</item>
+        <item>0</item>
+        <item>0</item>
+        <item>0</item>
+        <item>0</item>
+        <item>0</item>
+        <item>0</item>
     </integer-array>
 
+    <integer name="config_screenBrightnessDim">5</integer>
+
     <!-- Base "touch slop" value used by ViewConfiguration as a
          movement threshold where scrolling should begin. -->
     <dimen name="config_viewConfigurationTouchSlop">24dp</dimen>
@@ -188,4 +214,10 @@
     -->
     <bool name="config_enableWifiDisplay">true</bool>
 
+    <!-- Disables fading edges when marquee is enabled in TextView.
+         The fading edge effect will bring performance drop when
+	 the hardware rendering is used.
+    -->
+    <bool name="config_ui_enableFadingMarquee">false</bool>
+
 </resources>
diff --git a/sabresd_6dq/overlay/frameworks/base/core/res/res/values/config.xml b/sabresd_6dq/overlay/frameworks/base/core/res/res/values/config.xml
index edb7b3f..610a3f6 100755
--- a/sabresd_6dq/overlay/frameworks/base/core/res/res/values/config.xml
+++ b/sabresd_6dq/overlay/frameworks/base/core/res/res/values/config.xml
@@ -88,14 +88,18 @@
 
          Must be overridden in platform specific overlays -->
     <integer-array name="config_autoBrightnessLevels">
-        <item>10</item>
-        <item>40</item>
+        <item>5</item>
+        <item>15</item>
+        <item>50</item>
         <item>100</item>
-        <item>225</item>
-        <item>800</item>
-        <item>1600</item>
-        <item>3200</item>
-        <item>16000</item>
+        <item>200</item>
+        <item>400</item>
+        <item>1000</item>
+        <item>2000</item>
+        <item>3000</item>
+        <item>5000</item>
+        <item>10000</item>
+        <item>30000</item>
     </integer-array>
 
     <!-- Array of output values for LCD backlight corresponding to the LUX values
@@ -103,15 +107,19 @@
          than the size of the config_autoBrightnessLevels array.
          This must be overridden in platform specific overlays -->
     <integer-array name="config_autoBrightnessLcdBacklightValues">
-        <item>25</item>
-        <item>80</item>
-        <item>130</item>
-        <item>180</item>
-        <item>230</item>
-        <item>240</item>
-        <item>250</item>
-        <item>255</item>
-        <item>255</item>
+        <item>5</item>    <!-- 0-5 -->
+        <item>20</item>   <!-- 5-15 -->
+        <item>30</item>   <!-- 15-50 -->
+        <item>40</item>   <!-- 50-100 -->
+        <item>50</item>   <!-- 100-200 -->
+        <item>60</item>   <!-- 200-400 -->
+        <item>70</item>   <!-- 400-1000 -->
+        <item>80</item>   <!-- 1000-2000 -->
+        <item>130</item>  <!-- 2000-3000 -->
+        <item>180</item>  <!-- 3000-5000 -->
+        <item>255</item>  <!-- 5000-10000 -->
+        <item>255</item>  <!-- 10000-30000 -->
+        <item>255</item>  <!-- 30000+ -->
     </integer-array>
 
     <!-- Array of output values for button backlight corresponding to the LUX values
@@ -128,6 +136,14 @@
         <item>0</item>
         <item>0</item>
         <item>0</item>
+        <item>0</item>
+        <item>0</item>
+        <item>0</item>
+        <item>0</item>
+        <item>0</item>
+        <item>0</item>
+        <item>0</item>
+        <item>0</item>
     </integer-array>
 
     <!-- Array of output values for keyboard backlight corresponding to the LUX values
@@ -144,8 +160,18 @@
         <item>0</item>
         <item>0</item>
         <item>0</item>
+        <item>0</item>
+        <item>0</item>
+        <item>0</item>
+        <item>0</item>
+        <item>0</item>
+        <item>0</item>
+        <item>0</item>
+        <item>0</item>
     </integer-array>
 
+    <integer name="config_screenBrightnessDim">5</integer>
+
     <!-- Base "touch slop" value used by ViewConfiguration as a
          movement threshold where scrolling should begin. -->
     <dimen name="config_viewConfigurationTouchSlop">24dp</dimen>
-- 
1.8.0

